Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/repository/ITokenRepository.java

```diff
@@ -0,0 +1,22 @@
+package edu.asu.conceptpower.app.repository;
+
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.asu.conceptpower.app.model.Token;
+
+/**
+ * Token repository
+ *  
+ * @author Keerthivasan Krishnamurthy
+ * 
+ */
+@Repository
+public interface  ITokenRepository extends PagingAndSortingRepository<Token, String>{
+    
+    Token findByToken(String token); 
+    
+    @Transactional
```

Comment:
I don't think that'll do much here. Transactional annotation should be on the manager classes.

Reply:
I included the @Transactional annotation for the calling method as well. Shouldn't that be enough to handle the transactional failure? 

In this file - https://github.com/diging/conceptpower/blob/feature/CCP-248-2/Conceptpower%2BSpring/src/main/java/edu/asu/conceptpower/app/users/UserDatabaseClient.java

Reply:
You only need one transactional annotation (or multiple if you do several methods in a class, but as a class annotation makes in most cases sense). If you have it on the manager class, you don't need it here.